### DISCUSSION TITLE
score_cutoff ?

### DISCUSSION THREAD

**Discussion by shubhamscifi** [rapidfuzz/RapidFuzz#209] [Q&A] [answered]
The output should not be None in case 1 as the score_cutoff matches the fuzz.ratio. Why is it returning None in case1 but not in case2?
```
# case 1
a = "bc"
b = "bca"
print(rapidfuzz.fuzz.ratio(a, b))   # 80.0
print(process.extractOne(a, [b], scorer = rapidfuzz.fuzz.ratio, score_cutoff = 80.0))   # None

# case 2
a = "abcd"
b = "abce"
print(rapidfuzz.fuzz.ratio(a, b))   # 75.0
print(process.extractOne(a, [b], scorer= rapidfuzz.fuzz.ratio, score_cutoff = 75.0))    # ('abce', 75.0, 0)
```

**Comment by maxbachmann** [comment, marked as answer]
Yes the output should not be None in this case. The reason for this issue is that I perform some calculations using the score_cutoff internally to figure out thresholds to quit execution early. I currently do not take into account the imprecision caused by floating point math. I still need to think about how to address this in an optimal way.

For now a workaround would be to select a slightly lower value like `79.99`, which is not affected by this small imprecision. However this should definitely be fixed in RapidFuzz going forward.

**Reply by shubhamscifi** [reply]
Yeah. Thanks for the response. Should I make this discussion an issue?

**Reply by maxbachmann** [reply]
Yes this would probably help others with the same issue.
